src/Hpack/Config.hs

```diff
@@ -880,6 +885,7 @@ ensureRequiredCabalVersion inferredLicense pkg@Package{..} = pkg {
               uses "RebindableSyntax"
           && (uses "OverloadedStrings" || uses "OverloadedLists")
           && pathsModule `elem` getMentionedModules sect)
+      , makeVersion [2,2] <$ guard (sectionSatisfies (isJust . maybeSectionAConditional) sect)
```

Comment:
What this is saying is that if a nested conditional can be transformed into an `elif`, then require Cabal 2.2.

What I think you would want to do is the other way around: If the rendered `.cabal` file has `cabal-version: 2.2` (or later) then produce `elif`s.